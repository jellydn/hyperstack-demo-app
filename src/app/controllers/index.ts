// naming and imports: name and import as you see fit
import Auth from './auth'
import { NotesController } from './notes'

const controllers = []
controllers.push(Auth)
controllers.push(NotesController)

export const AppController = {
  controllers,
}
