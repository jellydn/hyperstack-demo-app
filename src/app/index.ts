import type { Context } from 'hyperstack'
import { context } from 'hyperstack'
import type * as Models from './models'
import type * as Mailers from './mailers'

export const appContext = context as Context<typeof Models, typeof Mailers>
