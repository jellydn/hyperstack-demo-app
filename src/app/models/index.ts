export { Note } from './note'
export { User } from './user'
