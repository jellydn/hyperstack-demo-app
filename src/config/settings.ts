// general purpose settings, with no infrastructure loading, no assumptions. keep it simple.
import path from 'path'
export const root = path.join(__dirname, '..')
