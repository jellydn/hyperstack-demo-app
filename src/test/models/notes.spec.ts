import { test } from '@hyperstackjs/testing'
import L from 'lodash'
import { root } from '../../config/settings'
import { appContext } from '../../app'
import { baseFixture } from '../__fixtures__'

const { models } = test(root)

const serializeModel = (m: any) => {
  return L.omit(m.toJSON(), ['pid', 'ownerId'])
}

describe('models', () => {
  describe('note', () => {
    models('should create with owner', async (_app) => {
      const { alex } = await baseFixture()
      const { Note } = appContext.models()
      const note = await Note.createWithOwner(alex, {
        title: 'a note',
        content: 'this and that',
      })
      expect(note.ownerId).toEqual(alex.id)
      expect(serializeModel(note)).toMatchSnapshot()
    })
    models('should list by owner', async (_app) => {
      const { alex } = await baseFixture()
      const { Note } = appContext.models()
      const notes = await Note.allByOwner(alex)
      expect(notes.map(serializeModel)).toMatchSnapshot()
    })
    models('should find by owner', async (_app) => {
      const { alex, sammy, youReallyGotMe } = await baseFixture()
      const { Note } = appContext.models()
      const note = await Note.oneByOwner(sammy, { pid: youReallyGotMe.pid })
      expect(
        await Note.oneByOwner(alex, { pid: youReallyGotMe.pid })
      ).toBeNull()
      expect(serializeModel(note)).toMatchSnapshot()
    })
  })
})
