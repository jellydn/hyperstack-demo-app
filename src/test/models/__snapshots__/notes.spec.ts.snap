// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models note should create with owner 1`] = `
{
  "content": "this and that",
  "title": "a note",
}
`;

exports[`models note should find by owner 1`] = `
{
  "content": "you got me so I don't know what I'm doin'",
  "title": "you really got me",
}
`;

exports[`models note should list by owner 1`] = `
[
  {
    "content": "you've got to roll with the punches to get what's real",
    "title": "jump",
  },
]
`;
